packages/ses/scripts/hermes-test.js

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env node
+/* global process */
+
+import { existsSync, writeFileSync, mkdtempSync, readFileSync } from 'node:fs';
+import path from 'node:path';
+import { execSync } from 'node:child_process';
+import os from 'node:os';
+
+// Utility function to determine OS-specific directory and create Hermes runner
+function createHermesRunner() {
+  const platform = os.platform();
+  let osDir;
+
+  switch (platform) {
+    case 'linux':
+      osDir = 'linux64-bin';
+      break;
+    case 'darwin':
+      osDir = 'osx-bin';
+      break;
+    case 'win32':
+      osDir = 'win64-bin';
+      break;
+    default:
+      throw new Error(`Unsupported OS: ${platform}`);
+  }
+
+  const hermesc = path.join(
+    '..',
+    '..',
+    'node_modules',
+    'hermes-engine-cli',
+    osDir,
+    'hermesc',
+  );
+  const hermes = path.join(
+    '..',
+    '..',
+    'node_modules',
+    'hermes-engine-cli',
+    osDir,
+    'hermes',
+  );
+
+  // Add .exe extension for Windows
+  const hermescPath = platform === 'win32' ? `${hermesc}.exe` : hermesc;
+  const hermesPath = platform === 'win32' ? `${hermes}.exe` : hermes;
+
+  return {
+    version: () => {
+      execSync(`"${hermesPath}" --version`, {
+        stdio: 'inherit',
+      });
+    },
+    run: jsFile => {
+      const hbcFile = jsFile.replace(/\.js$/, '.hbc');
+
+      // Compile
+      try {
+        execSync(
+          `"${hermescPath}" "${jsFile}" -emit-binary -out "${hbcFile}"`,
+          {
+            stdio: 'pipe', // Capture output to detect compilation errors
+          },
+        );
+      } catch (error) {
+        throw new Error(`Compilation error: ${error.message}`);
+      }
+
+      // Execute
+      execSync(`"${hermesPath}" -b "${hbcFile}"`, {
+        stdio: 'inherit',
+      });
+    },
+  };
+}
+
+const hermesPrefixSES = readFileSync(
+  path.join(import.meta.dirname, '..', 'dist', 'ses-hermes.cjs'),
+  'utf8',
+);
+const tempDir = mkdtempSync(path.join(os.tmpdir(), 'hermes-test-'));
+
+const prepareTestFile = testFile => {
+  const testName = path.basename(testFile);
+  const distFile = path.join(tempDir, `${testName}`);
+
+  const testContent = readFileSync(testFile, 'utf8');
+  // Combine the code in a way that line numbers are preserved and therefore helpful
+  const combined = `${testContent}
+;;;;
+${hermesPrefixSES}
+;;;;
+
+var TEST;
+function test(name, fn) {
```

Comment:
Thanks to the test going in a function I can start the concatenation with it, so if errors are throws, the line numbers will be correct. SES gets concatenated after but the test runs synchronously after SES still.